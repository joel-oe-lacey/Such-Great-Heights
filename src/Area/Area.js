import React from 'react';
import './Area.scss'

const Area = (props) => {
    return (
        <div className='area-card'>
          <h2 className='short-name'>{props.shortName}</h2>
          <h3 className='area-name'>{props.name}</h3>
          <p className='area-description'>{props.description}</p>
        </div>
    )
}

export default Area;
