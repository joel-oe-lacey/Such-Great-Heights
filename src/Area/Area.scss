.area-card {
  height: 60%;
  width: 20%;
  margin: 10px;
  padding: 10px;
  display: grid;
  grid-template-rows: 5% 5% 40% 50%;;
  grid-template-areas:
    'short-name'
    'name'
    'description'
    'button';
  justify-items: center;
  align-items: center;
  grid-row-gap: 10px;
  font-family: 'Work Sans', sans-serif;
  background-color: #FFD6BA;
  border-radius: 5%;
  border: 2px solid #555B6E;
  box-shadow: 3px 3px #393b40;
}

.short-name {
  grid-area: short-name;
}

.area-name {
  grid-area: name;
}

.area-description {
  margin-top: auto;
  grid-area: description;
  // padding: 3%;
  padding-top: 6%;
  border-bottom: 2px solid #555B6E;
  border-top: 2px solid #555B6E;
  height: 150%;
  text-align: center;
  font-size: 1.2em;
}

.view-listings-btn {
  margin-top: 10%;
  grid-area: button;
  height: 35%;
  border-radius: 4px;
  font-family: 'Work Sans', sans-serif;
  font-weight: 700;
  font-size: 1em;
  margin-right: 2%;
  background-color: #FAF9F9;
  cursor: pointer;
}
