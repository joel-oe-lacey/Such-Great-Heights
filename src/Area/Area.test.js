import React from 'react';
import { render } from '@testing-library/react';
import Area from './Area.js';
import { shallow } from 'enzyme';

describe('Area', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( < Area /> );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot when there are props', () => {
    let props = {
      shortName: 'RiNo',
      name: 'River North',
      description: 'this is an area in denver',
      id: 5
    }
    wrapper = shallow(<Area {...props} />)
    expect(wrapper).toMatchSnapshot();
  })


})
