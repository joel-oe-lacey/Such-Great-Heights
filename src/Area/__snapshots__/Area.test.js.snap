// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Area should match the snapshot 1`] = `
<div
  className="area-card"
>
  <h2
    className="short-name"
  />
  <h3
    className="area-name"
  />
  <p
    className="area-description"
  />
  <NavLink
    to="/areas/undefined"
  >
    <button
      className="view-listings-btn"
    >
      View Listings
    </button>
  </NavLink>
</div>
`;

exports[`Area should match the snapshot when there are props 1`] = `
<div
  className="area-card"
>
  <h2
    className="short-name"
  >
    RiNo
  </h2>
  <h3
    className="area-name"
  >
    River North
  </h3>
  <p
    className="area-description"
  >
    this is an area in denver
  </p>
  <NavLink
    to="/areas/5"
  >
    <button
      className="view-listings-btn"
    >
      View Listings
    </button>
  </NavLink>
</div>
`;
